modules/authwindowslive/lib/Auth/Source/LiveID.php

```diff
@@ -96,38 +100,32 @@ public function finalStep(&$state) {
 			),
 		);
 
-		$result = \SimpleSAML\Utils\HTTP::fetch('https://consent.live.com/AccessToken.aspx', $context);
+		$result = \SimpleSAML\Utils\HTTP::fetch('https://login.microsoftonline.com/common/oauth2/v2.0/token', $context);
 
-		parse_str($result, $response);
+		$response = json_decode($result,true);
 
 		// error checking of $response to make sure we can proceed
-		if (!array_key_exists('wrap_access_token',$response))
-			throw new Exception('[' . $response['error_code'] . '] ' . $response['wrap_error_reason'] .
-				"\r\nNo wrap_access_token returned - cannot proceed\r\n" . $response['internal_info']);
+		if (!array_key_exists('access_token',$response))
+			throw new Exception('[' . $response['error'] . '] ' . $response['error_description'] .
+				"\r\nNo access_token returned - cannot proceed\r\n" . implode(', ', $response['error_codes']));
```

Comment:
Please, avoid control structures without braces, as they open up for bugs that are difficult to find later. Even if you are only having one line inside, surround it with braces.

Of course, I'm conscious that the existing code is doing exactly the same and you just modified a few things inside that, but we are trying to get rid of all the code smells that we already have, which are quite a few...